### PR TITLE
Do not require connection string

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@seanfarmar this is related to #664

**Review by mikeminutillo — APPROVED** [review]
Happy with the changes as they are. I wonder if we should provide some more guidance when there is missing configuration.

File: src/Transport/AzureStorageQueueInfrastructure.cs [hunk 1]
```diff
@@ -47,21 +47,33 @@ internal AzureStorageQueueInfrastructure(SettingsHolder settings, string connect
                 settings.Set(WellKnownConfigurationKeys.DelayedDelivery.EnableTimeoutManager, false);
             }
 
+            if (!settings.TryGet(out queueServiceClientProvider))
+            {
+                if (connectionString == null)
+                {
+                    throw new Exception("Either a connection string or a queue client has to be provided in the configuration.");
```
[the remark this answers is not in the document]
Comment (mikeminutillo) [reply src/Transport/AzureStorageQueueInfrastructure.cs:54]: I put my suggestions for this in #696

[the remark this answers is not in the document]
Comment (SzymonPobiega) [reply src/Transport/AzureStorageQueueInfrastructure.cs]: 👍 Approved and merged

File: src/Transport/AzureStorageQueueInfrastructure.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ internal AzureStorageQueueInfrastructure(SettingsHolder settings, string connect
             this.settings = settings;
             this.connectionString = connectionString;
 
-            if (IsPremiumEndpoint(this.connectionString))
+            if (connectionString != null && IsPremiumEndpoint(this.connectionString))
```
[the remark this answers is not in the document]
Comment (SzymonPobiega) [reply src/Transport/AzureStorageQueueInfrastructure.cs:26]: There is a guard for that in the `ConnectionString` method. Both empty string and whitespace cause an exception at configuration time so null check here is enough.

**Pull request merged**